pdns/dnsdist-tcp.cc

```diff
@@ -897,6 +897,10 @@ static void handleQuery(std::shared_ptr<IncomingTCPConnectionState>& state, stru
   gettime(&queryRealTime, true);
 
   auto query = reinterpret_cast<char*>(&state->d_buffer.at(0));
+  time_t* cachedValidTime = new time_t(-1);
```

Comment:
This allocation doesn't get freed so it leaks.
Please consider to simply use time_t on the stack and pass a pointer to it to get it by reference, or better yet - use a C++ reference.